--- /dev/null
+++ ../src/Blocks/BlocksCli.php
@@ -10,6 +10,7 @@
 
 namespace MythemezVendor\EightshiftLibs\Blocks;
 
+use DirectoryIterator;
 use MythemezVendor\EightshiftLibs\Cli\AbstractCli;
 
 /**
@@ -139,14 +140,27 @@
 
 		\WP_CLI::runcommand("{$this->commandParentName} use_wrapper {$this->prepareArgsManual($args)}");
 
-		foreach (static::COMPONENTS as $component) {
+		foreach ($this->getElementNames('components') as $component) {
 			\WP_CLI::runcommand("{$this->commandParentName} use_component --name='{$component}' {$this->prepareArgsManual($args)}");
 		}
 
-		foreach (static::BLOCKS as $block) {
+		foreach ($this->getElementNames('custom') as $block) {
 			\WP_CLI::runcommand("{$this->commandParentName} use_block --name='{$block}' {$this->prepareArgsManual($args)}");
 		}
 
 		\WP_CLI::success('Blocks successfully set.');
+	}
+
+	public function getElementNames($type = 'components')
+	{
+		$elements = [];
+		$dir = new DirectoryIterator($this->getFrontendLibsBlockPath() . '/src/Blocks/' . $type);
+		foreach ($dir as $fileinfo) {
+			if ($fileinfo->isDir() && !$fileinfo->isDot()) {
+				$elements[] = $fileinfo->getFilename();
+			}
+		}
+
+		return $elements;
 	}
 }
